packages/cactus-plugin-ledger-connector-corda/src/main-server/kotlin/gen/kotlin-spring/src/main/kotlin/org/hyperledger/cactus/plugin/ledger/connector/corda/server/impl/ApiPluginLedgerConnectorCordaServiceImpl.kt

```diff
@@ -40,6 +43,9 @@ class ApiPluginLedgerConnectorCordaServiceImpl(
     val rpc: NodeRPCConnection
 ) : ApiPluginLedgerConnectorCordaService {
 
+    val watchedTransactionsBuffer = mutableMapOf<String, MutableList<GetMonitorTransactionsV1ResponseTx>>()
```

Comment:
@outSH How will this work if the connector is deployed behind a load balancer in an auto-scaling group? Can it be made work with sticky sessions for example?

Reply:
@petermetz What do you mean, isn't sticky session configured on a load-balancer only? How can we make an application work with that?

I will add multi-user support soon to this PR, but even then a client will have to call a single server I'm afraid.

Reply:
@outSH My question was not phrased very well, apologies. I'll try differently: 
Is there (or is there going to be) some recovery mechanism if the server instance goes away or is the client expected to implement this logic for now?

`Client -> LB -> [A, B, C, D]`

If the client has the subscription with `A` but then it goes away and the `LB` migrates the sticky session to `B` and then `B` would start returning an empty list in the GetMonitorTransactionsV1 responses (if I read the code right, but I'm not a Kotlin pro myself either)

Reply:
@petermetz Yeah, that make sense. The initial implementation assumed single connector and single client calling it (our current requirements for this feature are very basic), so the case you've described was not supported. I've done a major rework and reorganization of kotlin code. One of the changes is support for multiple clients. Fetching transactions from server on which we didn't explicitly start the monitoring first will end up with an error now, so i think the scenario you've described is solved now (the caller will be informed that the monitoring has broke and it needs to be restarted, possibly on new connector node).

Reply:
@outSH Awesome, marking this as resolved, thank you for the updates!